packages/repository/src/repositories/inclusion.ts

```diff
@@ -0,0 +1,96 @@
+import {Entity} from '../model';
+import {Getter} from '@loopback/core';
+import {
+  Filter,
+  Where,
+  resolveHasManyMetadata,
+  resolveBelongsToMetadata,
+  RelationMetadata,
+  RelationType,
+  constrainWhere,
+} from '../';
+import {AnyObject} from '..';
+import {DefaultCrudRepository} from './legacy-juggler-bridge';
+import {inspect} from 'util';
+import {
+  HasManyDefinition,
+  BelongsToDefinition,
+  HasManyResolvedDefinition,
+  BelongsToResolvedDefinition,
+} from '../relations';
+
+type ResolvedRelationMetadata =
+  | HasManyResolvedDefinition
+  | BelongsToResolvedDefinition;
+
+// SE: the source entity
+// TE: the target entity
+// SID: the ID of source entity
+// TID: the ID of target entity
+
+export class InclusionHandler<SE extends Entity, SID> {
```

Comment:
The name is a bit confusing as `InclusionHandler` contains a list of `_handlers`.

Reply:
yeah it should be plural